src/lightning_app/utilities/packaging/build_config.py

```diff
@@ -101,76 +106,90 @@ def build_commands(self):
         """
         return []
 
-    def on_work_init(self, work, cloud_compute: Optional["CloudCompute"] = None):
+    def on_work_init(self, work: "L.LightningWork", cloud_compute: Optional["CloudCompute"] = None) -> None:
         """Override with your own logic to load the requirements or dockerfile."""
-        try:
-            self.requirements = sorted(self.requirements or self._find_requirements(work) or [])
-            self.dockerfile = self.dockerfile or self._find_dockerfile(work)
-        except TypeError:
-            logger.debug("The provided work couldn't be found.")
+        found_requirements = self._find_requirements(work)
+        if self.requirements:
+            if found_requirements and self.requirements != found_requirements:
+                # notify the user of this silent behaviour
+                logger.info(
+                    f"A 'requirements.txt' exists with {found_requirements} but {self.requirements} was passed to"
+                    f" the `{type(self).__name__}`. The `requirements.txt` file will be ignored."
+                )
```

Comment:
I believe it would be more useful to show the work name rather than just the build config class name.

```suggestion
                logger.info(
                    f"A 'requirements.txt' exists with {found_requirements} but {self.requirements} was passed to"
                    f" the `{type(self).__name__}` in {work.name}. The `requirements.txt` file will be ignored."
                )
```

This way the user can identify the correct work among their X different work classes (maybe they are even using components not written by themselves).

Reply:
Thanks! TIL